src/shared/src/bitFlags.ts

```diff
@@ -34,7 +44,113 @@ export class BitFlags {
         return (new Float64Array(Uint32Array.from(this.flags).buffer))[0];
     }
 
+    toDebug(): string {
+        const debug: string[] = [];
+        this.flags.forEach((flag, index) => {
+            debug.push(flag.toString(2));
+            const fL = 32 - flag.toString(2).length;
+            for (let i = 0; i < fL; i++) {
+                debug[index] = '0'.concat(debug[index]);
+            }
+        });
+        return (`HIGH [ ${debug[1]} | ${debug[0]} ] LOW`);
+    }
+
+    toNumber(): number {
+        return this.flags[1] * 2 ** 32 + this.flags[0];
+    }
+
     toString(): string {
-        return (`[ ${this.flags[0].toString(2)} | ${this.flags[1].toString(2)} ]`);
+        return this.toNumber().toString();
+    }
+
+    getTotalBits(): number {
+        let total = 0;
+        this.flags.forEach((flag) => {
+            const n = 32;
+            let i = 0;
+            while (i++ < n) {
+                if ((1 << i & flag) === (1 << i)) {
+                    total++;
+                }
+            }
+        });
+        return total;
+    }
+}
+
+export class SeatFlags extends BitFlags {
```

Comment:
There should not be any payload or other system specific logic in a generic utility class file.

Reply:
This is not system specific to only weight/balance as other systems that need to read passenger seating information will use this utility class.

Reply:
Would be better to create a file for the loading specific stuff rather than stuffing it in bitFlags.ts.